### PR TITLE
Fix TimeUtils months_old and years_old returning incorrect values

### LINKED ISSUE #94: time_utils refinement: months_old and years_old methods are broken
## Problem

The time utilities refinement defines convenience methods `months_old` and `years_old` that call `age_in(:months)` and `age_in(:years)` respectively, but the underlying `age_in` method doesn't support these units.

**Location**: `lib/familia/refinements/time_utils.rb` lines 181-182

## Current Code

```ruby
# These methods are defined:
def months_old(*) = age_in(:months, *)
def years_old(*) = age_in(:years, *)

# But age_in only handles these cases:
def age_in(unit, from_time = nil)
  case UNIT_METHODS.fetch(unit.to_s.downcase, nil)
  when :days then age_seconds / PER_DAY
  when :hours then age_seconds / PER_HOUR  
  when :minutes then age_seconds / PER_MINUTE
  when :weeks then age_seconds / PER_WEEK
  else age_seconds  # ← months and years fall through to this
  end
end
```

## Impact

- `timestamp.months_old` returns age in seconds (not months)
- `timestamp.years_old` returns age in seconds (not years)
- These methods appear to work but return incorrect values

## Possible Solutions

1. **Add missing cases** to `age_in` method:
   ```ruby
   when :months then age_seconds / (PER_DAY * 30.44)  # Average month
   when :years then age_seconds / PER_YEAR
   ```

2. **Remove unsupported methods** (`months_old`, `years_old`)

3. **Document the limitation** if months/years are intentionally unsupported

## Note

The `UNIT_METHODS` hash already maps `'months'` → `:months` and `'years'` → `:years`, and `PER_YEAR` constant exists, suggesting months/years were intended to be supported.

## Discovered By

GitHub Copilot code review

### PR DESCRIPTION
**Pull request by delano**
## Summary

- Fixed `months_old` and `years_old` methods in TimeUtils refinement returning raw seconds instead of proper age values
- Added comprehensive month support with accurate calculations using 30.44 days per month
- Added extensive test coverage for all TimeUtils age calculation methods

## Changes Made

- **Added `PER_MONTH` constant**: 2,629,746 seconds (30.44 days, accounting for leap years)
- **Fixed `age_in` method**: Added proper handling for `:months` and `:years` cases
- **Enhanced `UNIT_METHODS`**: Added month mappings (`'mo'`, `'month'`, `'months'`)
- **Extended Numeric refinement**: Added `months`, `month`, `in_months` methods
- **Updated `in_seconds` method**: Added `:months` case for complete support

## Test Coverage

- Created `try/core/time_utils_try.rb` with 21 comprehensive test cases
- All existing tests continue to pass (1659 total test cases)
- Verified fix resolves the original issue where methods returned millions of seconds

## Before/After

**Before**: `timestamp.months_old` returned `2629746` (raw seconds)  
**After**: `timestamp.months_old` returns `1.0` (correct months)

**Before**: `timestamp.years_old` returned `31536000` (raw seconds)  
**After**: `timestamp.years_old` returns `1.0` (correct years)

Closes #94

🤖 Generated with [Claude Code](https://claude.ai/code)

___

### **PR Type**
Bug fix, Enhancement, Tests

___

### **Description**
- Fixed `months_old` and `years_old` methods returning raw seconds instead of proper age values

- Added comprehensive month support with `PER_MONTH` constant and conversion methods

- Enhanced `age_in` method to handle `:months` and `:years` cases properly

- Added extensive test coverage with 21 test cases for TimeUtils functionality

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["TimeUtils Bug"] --> B["Add PER_MONTH constant"]
  B --> C["Fix age_in method"]
  C --> D["Add month conversion methods"]
  D --> E["Update UNIT_METHODS mapping"]
  E --> F["Add comprehensive tests"]
  F --> G["Fixed months_old/years_old"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>time_utils.rb</strong><dd><code>Fix TimeUtils age calculations and add month support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/familia/refinements/time_utils.rb

<ul><li>Added <code>PER_MONTH</code> constant (2,629,746 seconds for 30.44 days)<br> <li> Enhanced <code>age_in</code> method with <code>:months</code> and <code>:years</code> case handling<br> <li> Added month conversion methods (<code>months</code>, <code>month</code>, <code>in_months</code>)<br> <li> Extended <code>UNIT_METHODS</code> hash with month mappings</ul>

</details>

  </td>
  <td><a href="https://github.com/delano/familia/pull/96/files#diff-8960f8c1c09fab3882693ae8a009e147a85f1df87c3d0439e5801bbdf566d51a">+10/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>time_utils_try.rb</strong><dd><code>Add comprehensive TimeUtils test coverage</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

try/core/time_utils_try.rb

<ul><li>Created comprehensive test suite with 21 test cases<br> <li> Tests month/year conversion methods and age calculations<br> <li> Verifies bug fix for <code>months_old</code> and <code>years_old</code> methods<br> <li> Includes edge cases and parameter validation tests</ul>

</details>

  </td>
  <td><a href="https://github.com/delano/familia/pull/96/files#diff-ba6716add42902086524d85c9078d1b41b23e5d4040b7717b2ceaa5c419e151d">+119/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>20250907_150748_delano_94_time_utils_refinement_months_old_and_years_old_methods_are_broken.rst</strong><dd><code>Document TimeUtils bug fix and enhancements</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

changelog.d/20250907_150748_delano_94_time_utils_refinement_months_old_and_years_old_methods_are_broken.rst

<ul><li>Documents the bug fix for <code>months_old</code> and <code>years_old</code> methods<br> <li> Lists new features including <code>PER_MONTH</code> constant and conversion methods<br> <li> Credits AI assistance for implementation and testing</ul>

</details>

  </td>
  <td><a href="https://github.com/delano/familia/pull/96/files#diff-7cb8d54ceac96a51bea721a7028021358fc697cac2a94382013371e696733004">+18/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: try/core/time_utils_try.rb [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+## Numeric#in_months - convert seconds to months
+RefinedContext.instance_eval_in_refined_context("2629746.in_months")
+#=> 1.0
+
+## Numeric#in_years - convert seconds to years
+RefinedContext.eval_in_refined_context("31536000.in_years")
+#=> 1.0
+
+## String#in_seconds - parse month string
+RefinedContext.instance_eval_in_refined_context("'1mo'.in_seconds")
```
[the remark this answers is not in the document]
Comment (delano) [reply try/core/time_utils_try.rb:42]: It's intentional
